### PR TITLE
Allow for inet6 option

### LINKED ISSUE #78: Is IPv6 supported?
It seems like the `opts` given to `:gen_udp.open/2` maybe needs to have `:inet6` given to enable support for IPv6.   If so, and if I'm reading the [UDP module](https://github.com/beam-telemetry/telemetry_metrics_statsd/blob/6f0df1e21fbb68df05f8040f9d85c0e1c0a05c47/lib/telemetry_metrics_statsd/udp.ex) right, that's not currently possible.

Does that all seem right to you?  I could potentially make a PR for this, but I wanted to sanity check first 😄

### PR DESCRIPTION
**Pull request by cheerfulstoic**
Fixes #78

I'm using this to be able to send metrics over IPv6 on fly.io through WireGuard

### CODE REVIEW

**Comment by cadebward** [comment]
This doesn't seem to work for me. `:inet.gethostbyname` still fails when looking up a host on ipv6.

I think it needs to be updated to also take in the `:inet6` option here https://github.com/beam-telemetry/telemetry_metrics_statsd/blob/main/lib/telemetry_metrics_statsd.ex#L547 (and a few lines up).

`:inet.gethostbyname(host, :inet6)` seems to work

**Comment by cheerfulstoic** [comment]
👍 Thanks for pointing this out!  I was using IPv6 addresses directly instead of a hostname so that's probably why I didn't run into this.

To keep things cleaner I took away the `inet6` option and replaced it with a `inet_address_family` option which can be either `inet` (the default), `inet6`, or `local` which is what is used by the [`inet:address_family()` type](https://www.erlang.org/doc/man/inet.html#type-address_family).  `:udp.open/2` doesn't use `inet:address_family()` but rather allows specifying an atom for the option that you want to use, so I adapted to use that as best as I could.

I'm on holiday right now, so I won't get back to testing this right away, but I can test it with my project soon-ish.

**Comment by cadebward** [comment]
I tested it out, works great!

**Comment by cheerfulstoic** [comment]
I forgot to update that I've been using this change in my project for a week or so now and it's been working fine 👍

File: lib/telemetry_metrics_statsd/options.ex [hunk 1]
```diff
@@ -21,6 +21,11 @@ defmodule TelemetryMetricsStatsd.Options do
       default: 8125,
       doc: "Port of the StatsD server."
     ],
+    inet_address_family: [
+      type: {:in, [:inet, :inet6, :local]},
+      default: :inet,
+      doc: "The inet address family, as specified by the Erlang inet:address_family type"
```
Comment (arkgil) [inline lib/telemetry_metrics_statsd/options.ex:27]:
```suggestion
      doc: "The inet address family, as specified by the Erlang `:inet.address_family type()`."
```

Comment (cheerfulstoic) [reply lib/telemetry_metrics_statsd/options.ex]: 👍  Fixed

**Comment by cheerfulstoic** [comment]
Saw that CI failed.  Just ran `mix format` which should hopefully help some.  Not sure what the other errors are...

**Comment by arkgil** [comment]
@cheerfulstoic we'll need to wait until #82 and #83 are merged in order to merge this one.

**Review by arkgil — COMMENTED** [review]
Thanks for the PR @cheerfulstoic! The code looks good, but we'll need a test for it too.

**Comment by cheerfulstoic** [comment]
👍  Sure, I can add a test.  I wasn't really sure how to do that, but I just dug in a bit.  I guess I would use the `given_udp_port_opened` / `assert_reported` helpers, but maybe adjust them a bit to allow for defining the address family.  Let me know if that seems off in some way

**Comment by cheerfulstoic** [comment]
I added a test, and I think it's probably the right thing, but let me know if you think it should work differently 😄

**Comment by arkgil** [comment]
@cheerfulstoic ideally we'd test that reporting works e2e with IPv6, so publishing a metric and using `assert_reported` in that test would be great!

**Comment by arkgil** [comment]
Also, if you could rebase on latest `main`, we could run tests for this PR. Thanks to @mopp CI is working again 💪

**Comment by cheerfulstoic** [comment]
Thanks very much!  I've added a separate test to match the pattern of the other tests (i.e. `:telemetry.executes` in the top-level tests and hostname resolutions inside of the `describe`).  It seems to fail when I disable my change

**Comment by cheerfulstoic** [comment]
(oh, and I rebased 👍 )

**Comment by arkgil** [comment]
Thank you for the contribution @cheerfulstoic. This change has been published in version 0.7.0 🎉

**Pull request merged**